core/server/middleware/theme-handler.js

```diff
@@ -60,6 +60,8 @@ themeHandler = {
 
         // clear the view cache
         blogApp.cache = {};
+        // reset the asset hash
+        config.assetHash = null;
```

Comment:
There are 3 cases:
1. you activate a theme 
2. you upload and activate a theme with a new name
3. you upload and activate a theme with the same name (theme override)

The two lines & all bug fixes so far have focused on case 3. This fix is for case 1 and I think also needed for case 2.